tests/test_model.py

```diff
@@ -1877,18 +1849,17 @@ def test_calculate_work_does_nothing_unless_edition_has_title_and_author(self):
         work, created = pool.calculate_work()
         eq_(True, created)
 
-        # Even before the forthcoming commit, the edition is clearly
-        # associated with the work.
+        # Even before the forthcoming commit, the edition is the
+        # work's presentation edition.
         eq_(work, edition.work)
-        eq_(True, edition.is_primary_for_work)
-        eq_(edition, work.primary_edition)
+        eq_(edition, work.presentation_edition)
 
-        # But without this commit, the join for the .primary_edition
+        # But without this commit, the join for the .presentation_edition
         # won't succeed and work.title won't work.
         self._db.commit()
```

Comment:
This might not be necessary anymore; I remember the problem had something to do with the foreign key relationships and you've just simplified that.